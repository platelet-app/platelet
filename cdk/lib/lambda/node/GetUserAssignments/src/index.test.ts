import _ from "lodash";
import { jest, expect } from "@jest/globals";

jest.unstable_mockModule("@platelet-app/lambda", () => ({
    request: jest.fn(),
    errorCheck: jest.fn(),
}));

// must be imported before the handler
const lambda = await import("@platelet-app/lambda");
// import handler
const { handler } = await import("./index.js");

function setupFetchStub(data: any): () => Promise<Response> {
    return function fetchStub(): Promise<Response> {
        return new Promise((resolve) => {
            resolve({
                json: () =>
                    Promise.resolve({
                        data,
                    }),
            } as Response);
        });
    };
}

const fakeAssignments = _.range(0, 5).map((i) => ({
    id: i,
    _version: 2,
}));
const fakeAssignments2 = _.range(5, 10).map((i) => ({
    id: i,
    _version: 1,
}));

const fakeData = {
    getUser: {
        assignments: {
            items: fakeAssignments,
            nextToken: "something",
        },
    },
};
const fakeData2 = {
    getUser: {
        assignments: {
            items: fakeAssignments2,
            nextToken: null,
        },
    },
};

describe("GetUserAssignments", () => {
    test("get user assginments", async () => {
        lambda.request
            .mockImplementationOnce(setupFetchStub(fakeData))
            .mockImplementationOnce(setupFetchStub(fakeData2))
            .mockImplementation(setupFetchStub({}));
        await handler({ userId: "test", retryCount: 1 });
        expect(lambda.request.mock.calls).toMatchSnapshot();
    });
});
