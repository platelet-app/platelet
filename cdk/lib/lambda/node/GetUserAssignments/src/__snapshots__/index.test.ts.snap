// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GetUserAssignments get user assginments 1`] = `
[
  [
    {
      "query": "query GetUser($id: ID!, $nextToken: String) {
  getUser(id: $id) {
    id
    assignments(nextToken: $nextToken) {
      nextToken
      items {
          id
          _version
          _deleted
      }
    }
  }
}",
      "variables": {
        "id": "test",
        "nextToken": null,
      },
    },
    "https://api.example.com/graphql",
  ],
  [
    {
      "query": "query GetUser($id: ID!, $nextToken: String) {
  getUser(id: $id) {
    id
    assignments(nextToken: $nextToken) {
      nextToken
      items {
          id
          _version
          _deleted
      }
    }
  }
}",
      "variables": {
        "id": "test",
        "nextToken": "something",
      },
    },
    "https://api.example.com/graphql",
  ],
]
`;
